s3/src/s3_lib.py

```diff
@@ -0,0 +1,372 @@
+# Copyright 2025 Canonical Ltd.
```

Comment:
Great work on creating this new version of the lib. Let's also create the ticket for publishing this in CharmHub via `data-platform-libs` charm.